### DISCUSSION TITLE
How to organize posts to have nice and stable links?

### DISCUSSION THREAD

**Discussion by idnm** [quarto-dev/quarto-cli#5384] [Q&A]
Hi there! I recently migrated by blog to quarto and now wonder how to best organize the `posts` directory. Currently, I have something like this
posts/
  -post_where_I_discuss_A/
    -post_where_I_discuss_A.qmd
   -post_where_I_discuss_B/
    -post_where_I_discuss_B.qmd
  
My problem with this structure is
1. The link for posts looks like `blog/posts/post_where_I_discuss_A/post_where_I_discuss_A` which is unnecessarily repetitive. What is a good practice instead?
2. So far my few posts are all in the same listing, but what if I decide later to make several categories? I'd think I will make two separate folders like `posts_about_walking` and `posts_about_talking` and distribute the existing posts between them. But this will change the links to the posts, and I'd like them to be stable if I decide to restructure the blog. How to go about that?

Thanks!

**Comment by mcanouil** [comment]
What you want to do is either change your structure which is going to be used to create the default links or use aliases, see <https://quarto.org/docs/websites/website-navigation.html#redirects>.

Also naming your Quarto documents `index.qmd` will also avoid the duplicated part of the URI.
See for example <https://mickael.canouil.fr/posts/2023-03-12-quarto-mathjax-packages/> and the source code <https://github.com/mcanouil/mickael.canouil.fr/tree/main/posts/2023-03-12-quarto-mathjax-packages>.

**Reply by idnm** [reply]
Sorry, maybe this is a misguided question, but is there a way to bind a URL to the post itself, not to its location in the project?

Say I want to specify something like
----
title: post about A
url: https://blog/post_about_A
----

So that this URL is fixed even if I move the post around in the project directory or put in a different listing later?

Also, a bit tangential, but I'd appreciate an advice. Is specifying the date in the post title considered to be a good practice?

**Reply by mcanouil** [reply]
That's what redirects/aliases are for as described in my first reply.

If you want more control, you'll have to rely on standard HTML/web processes, _i.e._, `.htaccess`.
